services/gp-search/pyproject.toml

```
@@ -11,6 +11,8 @@ dependencies = [
     "fhir-resources (>=8.0.0,<9.0.0)",
     "pydantic (>=2.11.4,<3.0.0)",
     "fastjsonschema (>=2.21.1,<3.0.0)",
+    "aws-lambda-powertools (>=3.11.0,<3.13.0)",
+    "aws-xray-sdk (>=2.14.0,<3.0.0)",
```

Could we be explicit about powetools being version 3.12 like this

```suggestion
    "aws-lambda-powertools (>=3.12.0,<3.13.0)",
    "aws-xray-sdk (>=2.14.0,<3.0.0)",
```

And also we can remove these 2 dependencies from the dev.dependencies below

---

Also perform `poetry lock` command and commit the lock file changes (if any) when updating pyproject.toml